### PR TITLE
Create static file provider feature

### LINKED ISSUE #1221: Need to serve multi tenant supported static file
They are tenant specific.  Mostly at tenant `/`.

For example,
`/.well-known/apple-developer-merchantid-domain-association` 

Others would be
* robots.txt
* sitemaps files
* domain verifications files (like one required for google apps)
* app manifest files (manifest.json)
* etc...

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #1221

### CODE REVIEW

**Comment by Jetski5822** [comment]
LGTM

File: src/OrchardCore.Modules/OrchardCore.Media/Startup.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public override void ConfigureServices(IServiceCollection services)
                 var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
                 var env = serviceProvider.GetRequiredService<IHostingEnvironment>();
```
Comment (awyl) [inline src/OrchardCore.Modules/OrchardCore.Media/Startup.cs]: Remove?

**Comment by awyl** [comment]
Just have it compiled and tested.

It still doesn't serve the file right.  I got `The page could not be found.` for `/.well-known/apple-developer-merchantid-domain-association`.  If I rename it to `/.well-known/a.txt` it works.

**Comment by sebastienros** [comment]
Interesting, looks like a bug in the file provider, but I might be wrong, will check asap.

**Comment by petedavis** [comment]
This could also be used to implement lets encrypt and serve up the challenge files.

**Comment by sebastienros** [comment]
@petedavis I assign you the task to create a Let's Encrypt module then, it would create and configure the certificate, and also renew automatically. And by the way you can also port the SSL module from O1 ;)

**Comment by sebastienros** [comment]
@awyl fixed

**Comment by awyl** [comment]
Works!

Thanks so much!!!

**Comment by awyl** [comment]
Is it me or the module.txt should also specify the tenant module in features?

The tenant menu in admin disappeared and no where to be found in modules.

**Comment by sebastienros** [comment]
Probably, I thought it was not mandatory, but apparently it is as soon as you create another feature.

**Comment by sebastienros** [comment]
Done

**Comment by awyl** [comment]
Tested. Perfect.  Thanks!

File: src/OrchardCore.Modules/OrchardCore.Tenants/Module.txt [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
 OrchardVersion: 2.0.x
-Category: Infrastructure
\ No newline at end of file
+Features:
+    OrchardCore.Tenants:
+        Name: Tenants
+        Description: Provides a way to manage tenants from the admin.
+        Category: Infrastructure
+    OrchardCore.Tenants.FileProvider:
+        Name: Satic File Provider
```
Comment (agriffard) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Module.txt]: Satic => Static

**Pull request merged**